### PR TITLE
Implement `plotResidualsVsTime()` and `plotResidualsVsSimulated()`

### LINKED ISSUE #803: Add `$calculateResiduals()` method to `DataCombined` object
We will need to compute residuals between observed and simulated data for plotting. 

So, the question is: 

- Should we carry out this calculation only inside the plotting functions where residuals are needed? 

or

- Should the `DataCombined` object have a `$computeResiduals()` method that keeps up-to-date information on residuals in a dedicated column in `private$.dataCombined` dataframe?

I'd vote for the second option because this way the users have flexibility in terms of how and where they can use the residuals data. If, instead, we do this only inside the plotting functions, the users will not have access to this data directly. 

Also, in terms of implementation, I was thinking that `$computeResiduals()` will be a private method that updates residuals every time new datasets are added. Users can access this data, of course, through `toDataFrame()` public method.

(Note that the users will need this information regardless of whether they use _tlf_ for plotting or create custom plots.)

### LINKED ISSUE #925: Implement `plotResidualsVsTime()`
As discussed in https://github.com/Open-Systems-Pharmacology/OSPSuite-R/discussions/811
Related to #674

### LINKED ISSUE #1036: Implement `plotResidualsVsSimulated()`
As discussed in https://github.com/Open-Systems-Pharmacology/OSPSuite-R/discussions/811
Related to #674

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
TODO

- [x] Implement residuals calculation function
- [x] Implement `plotResidualsVsTime()`
- [x] Implement `plotResidualsVsSimulated()`
- [x] Tests for `plotResidualsVsTime()` 
- [x] Tests for `plotResidualsVsSimulated()` 
- [x] Refactor to avoid duplication between scatter plots
- [x] Refactor to avoid duplication across plotting functions
- [x] More tests for plotting utility functions
- [x] Check comments

Related to #674 

Discussed in https://github.com/Open-Systems-Pharmacology/OSPSuite-R/discussions/811

Closes #803 
Closes #925
Closes #1036

### CODE REVIEW

**Comment by IndrajeetPatil** [comment]
Here is what the default plot looks like:

```r
library(ospsuite)

# load the simulation
sim <- loadTestSimulation("MinimalModel")
simResults <- importResultsFromCSV(
  simulation = sim,
  filePaths = getTestDataFilePath("Stevens_2012_placebo_indiv_results.csv")
)

# import observed data (will return a list of DataSet objects)
dataSet <- loadDataSetsFromExcel(
  xlsFilePath = getTestDataFilePath("CompiledDataSetStevens2012.xlsx"),
  importerConfiguration = loadDataImporterConfiguration(getTestDataFilePath("ImporterConfiguration.xml"))
)

# create a new instance and add datasets
myCombDat <- DataCombined$new()
myCombDat$addDataSets(dataSet)
myCombDat$addSimulationResults(
  simResults,
  quantitiesOrPaths = c(
    "Organism|Lumen|Stomach|Metformin|Gastric retention",
    "Organism|Lumen|Stomach|Metformin|Gastric retention distal",
    "Organism|Lumen|Stomach|Metformin|Gastric retention proximal"
  )
)

myCombDat$setGroups(
  names = c(
    "Organism|Lumen|Stomach|Metformin|Gastric retention",
    "Organism|Lumen|Stomach|Metformin|Gastric retention distal",
    "Organism|Lumen|Stomach|Metformin|Gastric retention proximal",
    "Stevens_2012_placebo.Placebo_total",
    "Stevens_2012_placebo.Placebo_distal",
    "Stevens_2012_placebo.Placebo_proximal"
  ),
  groups = c("Solid total", "Solid distal", "Solid proximal", "Solid total", "Solid distal", "Solid proximal")
)

# create plot config
myPC <- DefaultPlotConfiguration$new()
myPC$legendPosition <- tlf::LegendPositions$outsideRight

# plot
plotResidualsVsTime(myCombDat, myPC)
```

obs_vs_pred

**Comment by IndrajeetPatil** [comment]
And here is what the default `plotResidualsVsSimulated()` output looks like:

```r
library(ospsuite)

# load the simulation
sim <- loadTestSimulation("MinimalModel")
simResults <- importResultsFromCSV(
  simulation = sim,
  filePaths = getTestDataFilePath("Stevens_2012_placebo_indiv_results.csv")
)

# import observed data (will return a list of DataSet objects)
dataSet <- loadDataSetsFromExcel(
  xlsFilePath = getTestDataFilePath("CompiledDataSetStevens2012.xlsx"),
  importerConfiguration = loadDataImporterConfiguration(getTestDataFilePath("ImporterConfiguration.xml"))
)

# create a new instance and add datasets
myCombDat <- DataCombined$new()
myCombDat$addDataSets(dataSet)
myCombDat$addSimulationResults(
  simResults,
  quantitiesOrPaths = c(
    "Organism|Lumen|Stomach|Metformin|Gastric retention",
    "Organism|Lumen|Stomach|Metformin|Gastric retention distal",
    "Organism|Lumen|Stomach|Metformin|Gastric retention proximal"
  )
)

myCombDat$setGroups(
  names = c(
    "Organism|Lumen|Stomach|Metformin|Gastric retention",
    "Organism|Lumen|Stomach|Metformin|Gastric retention distal",
    "Organism|Lumen|Stomach|Metformin|Gastric retention proximal",
    "Stevens_2012_placebo.Placebo_total",
    "Stevens_2012_placebo.Placebo_distal",
    "Stevens_2012_placebo.Placebo_proximal"
  ),
  groups = c("Solid total", "Solid distal", "Solid proximal", "Solid total", "Solid distal", "Solid proximal")
)

# default
plotResidualsVsSimulated(myCombDat)
```

obs_vs_pred

**Comment by PavelBal** [comment]
> Here is what the default plotResidualsVsTime() output looks like:

X-axis values (or units) can't be right ;)

File: NEWS.md
Comment (PavelBal) [inline NEWS.md:5]: "for observed versus simulated..." to be consistent with function naming.

File: NEWS.md
Comment (PavelBal) [inline NEWS.md:7]: as above - simulated instead of predicted.

**Review by PavelBal — CHANGES_REQUESTED** [review]
Small changes.

I did not review the code, just from the user perspective.

**Comment by PavelBal** [comment]
Could you add to this commit to NEWS.MD:

- The class `SimulationBatch` gets a new property `id`
- The output of `runSimulationBatches()` is now a named list with names being the ids of `SimulationBatch`

**Comment by IndrajeetPatil** [comment]
Okay, updated NEWS to address all of the above-mentioned points.

**Review by PavelBal — APPROVED** [review]

**Comment by PavelBal** [comment]
Please let someone else review the code before merging (until for a while).

**Comment by IndrajeetPatil** [comment]
Yepp, we do need at least 2 approvals on this one.

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**
